examples/ShoppingAds/AddShoppingSmartAd.php

```diff
@@ -0,0 +1,424 @@
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $budgetResourceName,
+        int $merchantCenterAccountId
+    ) {
+        // Configures the shopping settings for Smart Shopping campaigns.
+        $shoppingSettings = new ShoppingSetting([
+            // Sets the sales country of products to include in the campaign.
+            // Only products from Merchant Center targeting this country will appear in the
+            // campaign.
+            'sales_country' => new StringValue(['value' => 'US']),
+            'merchant_id' => new Int64Value(['value' => $merchantCenterAccountId])
+        ]);
+
+        // Creates the campaign.
+        $campaign = new Campaign([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise Campaign #' . uniqid()]),
+            // Configures settings related to shopping campaigns including
+            // advertising channel type, advertising sub-type and shopping setting.
+            'advertising_channel_type' => AdvertisingChannelType::SHOPPING,
+            'advertising_channel_sub_type' => AdvertisingChannelSubType::SHOPPING_SMART_ADS,
+            'shopping_setting' => $shoppingSettings,
+            // Recommendation: Set the campaign to PAUSED when creating it to prevent
+            // the ads from immediately serving. Set to ENABLED once you've added
+            // targeting and the ads are ready to serve.
+            'status' => CampaignStatus::PAUSED,
+            // Bidding strategy must be set directly on the campaign.
+            // Setting a portfolio bidding strategy by resource name is not supported.
+            // Maximize conversion value is the only strategy supported for Smart Shopping
+            // campaigns.
+            // An optional ROAS (Return on Advertising Spend) can be set for
+            // MaximizeConversionValue.
+            // The ROAS value must be specified as a ratio in the API. It is calculated by dividing
+            // "total value" by "total spend".
+            // For more information on maximize conversion value, see the support article:
+            // http://support.google.com/google-ads/answer/7684216.
+            'maximize_conversion_value' => new MaximizeConversionValue([
+                'target_roas' => new DoubleValue(['value' => 3.5])
+            ]),
+            // Sets the budget.
+            'campaign_budget' => new StringValue(['value' => $budgetResourceName])
+        ]);
+
+        // Creates a campaign operation.
+        $campaignOperation = new CampaignOperation();
+        $campaignOperation->setCreate($campaign);
+
+        // Issues a mutate request to add the campaign.
+        $campaignServiceClient = $googleAdsClient->getCampaignServiceClient();
+        $response = $campaignServiceClient->mutateCampaigns($customerId, [$campaignOperation]);
+
+        /** @var Campaign $addedCampaign */
+        $addedCampaign = $response->getResults()[0];
```

Comment:
Nit: why don't you store the resource name in a variable so that you don't have to call getResourceName() twice?